SparseGrids/tsgGridWavelet.cpp

```diff
@@ -313,11 +332,27 @@ void GridWavelet::integrate(double q[], double *conformal_correction) const{
     }
 }
 
+void GridWavelet::differentiate(const double x[], double jacobian[]) const{
+    std::fill(jacobian, jacobian + num_outputs * num_dimensions, 0.0);
```

Comment:
* use `fill()` when you have two iterators, e.g., `fill(x.begin(), x.end(), 0.0)` or `fill(x, y, 1.0)`
* use `fill_n()` when you have a range, e.g., `fill_n(x, x + num, 0.0)` or `fill_n(x.begin(), x.begin() + num, 1.0)`
* do not user, `fill(x, x + num, 1.0)`

Reply:
* do not use `fill_n(x.begin(), x.size(), 1.0)`, prefer `fill(x.begin(), x.end(), 1.0)`

Reply:
Changed.